src/seedu/addressbook/commands/FindTagCommand.java

```diff
@@ -0,0 +1,59 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.tag.Tag;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds and lists all persons in address book whose tag contains any of the argument keywords.
+ * Keyword matching is case sensitive.
+ */
+public class FindTagCommand extends Command {
+    public static final String COMMAND_WORD = "findTag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose tags contain any of "
+            + "the specified keywords (case-sensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " teammates captain";
+
+    private Set<String> keywords;
+
+    public FindTagCommand(Set<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    public Set<String> getKeywords() {
+        return keywords;
+    }
+
+    @Override
+    public CommandResult execute() {
+        final List<ReadOnlyPerson> personsFound = getPersonsWithNameContainingAnyKeyword(keywords);
+        return new CommandResult(getMessageForPersonListShownSummary(personsFound), personsFound);
+    }
+
+    /**
+     * Retrieves all persons in the address book whose tags contain at least one of specified keywords.
+     *
+     * @param keywords for searching
+     * @return list of persons found
+     */
+    private List<ReadOnlyPerson> getPersonsWithNameContainingAnyKeyword(Set<String> keywords) {
```

Comment:
Perhaps you can change "Name" to "Tags" in your method name to make the method more easier to understand.
Aside from it, code is quite readable and simple to understand.